### ISSUE TITLE
decTitleKeys.bin isn't encrypted properly

### ISSUE THREAD

**Issue by shinji257** [d0k3/Decrypt9WIP#44]
When comparing the contents of an encTitleKeys.bin that was extracted directly from a 3ds vs one that was re-encrypted from a decTitleKeys.bin (also from the same 3ds) the contents do not match.  The former version actually aligns with the ticket.db file for encrypted title keys where as the latter seems to result in gibberish giving the wrong keys even for titles purchased from the eshop.

Keys were dumped using the print_ticket_keys.py/dump_ticket_keys.py using the encTitleKeys.bin and ticket.db respectively to check this and all files were dumped using the latest repository code.

**Comment by shinji257** [comment]
Thanks to Redy on IRC we identified the line.  I actually saw this myself earlier and found it odd

This line is the culprit.
https://github.com/d0k3/Decrypt9WIP/blob/master/source/decryptor/titlekey.c#L24

Right now it is actually decrypting already decrypted keys.  To encrypt it has to be set to AES_CNT_TITLEKEY_ENCRYPT_MODE in order to produce expected results.  I tested this myself by changing that bit (resulting in broken decryption for the test) and validating the results using an earlier decTitleKeys.bin that didn't generate a proper set of encrypted title keys before.  Now it generates a set that matches what would of been pulled from the device directly.
